File: Data Structures & Algorithms/longest-palindromic-substring/submission-5.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int max_len = 0;
        string ans ="";
        vector<vector<int>>dp(s.size(),vector<int>(s.size(),false));
        for(int i=s.size()-1;i>=0;--i){
            for(int j=i;j<s.size();++j){
                if(s[i]==s[j] and (j-i<=2 or dp[i+1][j-1]))dp[i][j]=true;
                if(dp[i][j]){
                    if(max_len<j-i+1){
                        ans = s.substr(i,j-i+1);
                        max_len = j-i+1;
                    }
                }
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/permutations/submission-5.py
class Solution:
    def go(self,nums, temp_nums, vis):
        if len(nums)==len(temp_nums):
            self.ans.append(temp_nums.copy())
            return 
        for i in range(len(nums)):
            if vis[i]==False:
                vis[i]=True
                temp_nums.append(nums[i])
                self.go(nums,temp_nums,vis)
                temp_nums.pop()
                vis[i]=False
        
    def permute(self, nums: List[int]) -> List[List[int]]:
        self.ans = []
        self.go(nums, [], [ False for _ in range(len(nums))] )
        return self.ans